pkg/virt-handler/ratelimitcmd/executor_test.go

```diff
@@ -0,0 +1,82 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package ratelimitcmd_test
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/util/clock"
+
+	"kubevirt.io/kubevirt/pkg/virt-handler/ratelimitcmd"
+)
+
+var _ = Describe("rate limited executor", func() {
+
+	var testsClock *clock.FakeClock
+
+	BeforeEach(func() {
+		testsClock = clock.NewFakeClock(time.Now())
+	})
+
+	It("should execute command when the underlying rate-limiter is not blocking", func() {
+		executor := ratelimitcmd.NewRateLimitedExecutor(newNonBlockingRateLimiter())
+		testsClock.Step(time.Nanosecond)
+
+		expectCommandExec(executor)
+		expectCommandExec(executor)
+	})
+
+	It("should not execute command when the underlying rate-limiter is blocking", func() {
+		executor := ratelimitcmd.NewRateLimitedExecutor(newBlockingRateLimiter())
+		testsClock.Step(time.Nanosecond)
+
+		expectSkipCommandExec(executor)
+		expectSkipCommandExec(executor)
+	})
+})
+
+func expectCommandExec(executor *ratelimitcmd.RateLimitedExecutor) {
+	ExpectWithOffset(1, executor.Exec(failingCommandStub())).To(MatchError(testsExecError))
+}
+
+func expectSkipCommandExec(executor *ratelimitcmd.RateLimitedExecutor) {
+	ExpectWithOffset(1, executor.Exec(failingCommandStub())).To(Succeed())
+}
+
+func newBlockingRateLimiter() *rateLimiterStub {
+	return &rateLimiterStub{block: true}
+}
+
+func newNonBlockingRateLimiter() *rateLimiterStub {
+	return &rateLimiterStub{block: false}
+}
```

Comment:
You do not really need these, you can just place the struct directly.

Reply:
Done